cfme/common/__init__.py

```diff
@@ -326,3 +368,88 @@ def validate_properties(self):
             assert expected_value == shown_value,\
                 ("Property '{}' has wrong value, expected '{}' but was '{}'"
                  .format(property_tuple, expected_value, shown_value))
+
+    def validate_tags(self, tags):
+        """Remove all tags and add `tags` from user input, validates added tags"""
+        self._validate_tags_internal()
+        tags_db_raw = self.get_tags(method='db')
+        if len(tags_db_raw) > 0:
+            self.remove_tags(tags=self._get_tags(db_tags=tags_db_raw))
+            tags_db_raw = self.get_tags(method='db')
+        assert len(tags_db_raw) == 0, "Some of tags still available in database!"
+        self.add_tags(tags)
+        self._validate_tags_internal(reference_tags=tags)
+
+    def _validate_tags_internal(self, tag="My Company Tags", reference_tags=None):
+        """
+        Validation method which check tagging between UI and database
+        To use this method, `seld` should be extended `Taggable` class
```

Comment:
```To use this method, `seld` should be extended `Taggable` class```
seld vs self. But anyway this sentence does not make much sense, I would reword it.